synapse/http/federation/matrix_federation_agent.py

```diff
@@ -109,6 +110,15 @@ def request(self, method, uri, headers=None, bodyProducer=None):
             else:
                 target = pick_server_from_list(server_list)
 
+        # make sure that the Host header is set correctly
+        if headers is None:
+            headers = Headers()
+        else:
+            headers = headers.copy()
+
+        if not headers.hasHeader(b'host'):
+            headers.addRawHeader(b'host', server_name_bytes)
```

Comment:
I'm assuming header names are case insensitive here?

Reply:
yeah, twisted sorts that out for us.